networkx/algorithms/tests/test_swap.py

```diff
@@ -5,6 +5,24 @@
 # random.seed(0)
 
 
+def test_directed_edge_swap():
+    graph = nx.gnp_random_graph(200, 0.5, directed=True)
+    in_degrees = sorted((n, d) for n, d in graph.in_degree())
+    out_degrees = sorted((n, d) for n, d in graph.out_degree())
+    G = nx.directed_edge_swap(graph, nswap=40, max_tries=500)
```

Comment:
Any tests that involve random numbers can be very difficult to track if they start failing. The test fails, and its hard to know why and then rerunning the test passes it again. So our policy has been to use a seed for the tests we are going to run regularly. This will be needed for both graph creation (`gnp_random_graph`) and for the swap. I guess this also means we can eliminate the test which explicitly tests that `seed` works. :}

Reply:
Makes sense! I went ahead and converted `test_directed_edge_swap` to use a path graph, added a seed to its `directed_edge_swap` call, and removed the seed test